src/checkovInstaller.ts

```diff
@@ -12,44 +15,145 @@ const isCheckovInstalledGlobally = async () => {
     }
 };
 
-const installOrUpdateCheckovWithPip3 = async (logger: Logger): Promise<string | null> => {
+const validateCheckovVersion = (checkovVersion: string | undefined) => {
+    // validates that the version is >= the min version (or it can be null)
+    // does NOT validate that the version actually exists (this will happen during the actual installation)
+    // assumes the value came from getCheckovVersion helper method
+    // a normal return indicates the version is valid, otherwise an error is thrown
+
+    if (!checkovVersion) {
+        return;
+    }
+
+    const parts = checkovVersion.split('.').map(i => parseInt(i));
+    if (parts.length !== 3) {
+        throw new Error(`Invalid specified checkov version: ${checkovVersion} (expected format #.#.#)`);
+    }
+
+    const minVersion = minCheckovVersion.split('.').map(i => parseInt(i));
+
+    let aboveMinimum = true;
+
+    for (let index = 0; index < 3; index++) {
+        // Validate left to right until we find a mismatch (also make sure we parsed actual numbers)
+        if (isNaN(parts[index])) {
+            throw new Error(`Invalid specified checkov version ${checkovVersion} (expected format #.#.#)`);
+        }
+
+        if (parts[index] > minVersion[index]) {
+            // all previous parts were ==, so this means we are above the minimum
+            break;
+        }
+        else if (parts[index] < minVersion[index]) {
+            // all previous parts were ==, so this means we are below the minimum
+            aboveMinimum = false;
+            break;
+        }
+    }
+
+    if (!aboveMinimum) {
+        throw new Error(`Checkov version ${checkovVersion} is below the minimum supported version of ${minCheckovVersion}`);
+    }
+};
+
+const installOrUpdateCheckovWithPip3 = async (logger: Logger, autoUpdate: boolean, checkovVersion: string | undefined): Promise<string | null> => {
+    logger.info('Trying to install Checkov using pip3.');
+
+    // Pip version logic:
+    // if a version is specified, install that ('pip install checkov==1.2.3' will remove any other version)
+    // otherwise, if autoupdate is true, then run with -U and no version
+    // otherwise just run 'pip install checkov', which will do nothing (other than validate that the pip3 installation works) 
+    // if any version is installed.
+
     try {
-        logger.info('Trying to install Checkov using pip3.');
-        await asyncExec('pip3 install -U --user --verbose checkov -i https://pypi.org/simple/');
+        let command = 'pip3 install --user --verbose ';
```

Comment:
If you agree on using a single variable we can build this install command in an immutable and shorter way (simple backticks string build \`${}\` 👌